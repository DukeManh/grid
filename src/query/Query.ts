import { IQueryAction, QueryAction } from './QueryAction';

interface IQuery {
  from: (table: string, schema?: string) => IQueryAction;
}

export class Query implements IQuery {
  from(table: string, schema?: string) {
    return new QueryAction({ name: table, schema: schema ?? 'public' });
  }
}
