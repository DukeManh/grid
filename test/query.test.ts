import Query from '../src/query';

const query = new Query();

test('basic select table', async () => {
  const res = query.from('users').select().toSql();
  expect(res).toMatchSnapshot();
});
test('select table with custom schema', async () => {
  const res = query.from('users', 'ext').select().toSql();
  expect(res).toMatchSnapshot();
});
test('select table with Capital name, schema', async () => {
  const res = query.from('UsErs', 'Ext').select().toSql();
  expect(res).toMatchSnapshot();
});
test('select table with columns', async () => {
  const res = query.from('users').select(['name', 'id', 'age']).toSql();
  expect(res).toMatchSnapshot();
});
test('select table with Capital columns', async () => {
  const res = query.from('users').select(['Name', 'id', 'Age']).toSql();
  expect(res).toMatchSnapshot();
});
test('select table with pagination', async () => {
  const res = query.from('users').select().range(0, 99).toSql();
  expect(res).toMatchSnapshot();
});
