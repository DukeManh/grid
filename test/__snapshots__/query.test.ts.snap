// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic select table 1`] = `"select * from public.users"`;

exports[`select table with Capital columns 1`] = `"select \\"Name\\",id,\\"Age\\" from public.users"`;

exports[`select table with Capital name, schema 1`] = `"select * from \\"Ext\\".\\"UsErs\\""`;

exports[`select table with columns 1`] = `"select name,id,age from public.users"`;

exports[`select table with custom schema 1`] = `"select * from ext.users"`;

exports[`select table with pagination 1`] = `"select * from public.users limit 100 offset 0"`;
